R/utils.R

```diff
@@ -330,6 +350,249 @@ markdown_to_latex <- function(text) {
     unname()
 }
 
+cmark_rules <- list(
+
+  heading = function(x, process) {
+    heading_sizes <- c(36, 32, 28, 24, 20, 16)
+    fs <- heading_sizes[as.numeric(xml2::xml_attr(x, attr = "level"))]
+
+    rtf_paste0(
+      rtf_raw("{\\ql \\f0 \\sa180 \\b \\fs"),
+      rtf_raw(fs),
+      rtf_raw(" "),
+      process(xml2::xml_children(x)),
+      rtf_raw("}")
+    )
+  },
+  thematic_break = function(x, process) {
+    rtf_raw("{\\qc \\f0 \\sa180 \\emdash\\emdash\\emdash\\emdash\\emdash}")
+  },
+  link = function(x, process) {
+    # NOTE: RTF doesn't handle the `title` attr (Pandoc also ignores)
+    destination <- xml2::xml_attr(x, attr = "destination")
+    text <- process(xml2::xml_children(x))
+    if (text == "") text <- destination
+    rtf_raw("{\\field{\\*\\fldinst{HYPERLINK \"", destination, "\"}}{\\fldrslt{\\ul ", text, "}}}")
+  },
+  list = function(x, process) {
+
+    type <- xml2::xml_attr(x, attr = "type")
+    n_items <- length(xml2::xml_children(x))
+
+    # NOTE: `start`, `delim`, and `tight` attrs are ignored; we also
+    # assume there is only `type` values of "ordered" and "bullet" (unordered)
+    rtf_raw(
+      paste(
+        "\\pard\\intbl\\itap1\\tx220\\tx720\\tx1133\\tx1700\\tx2267\\tx2834\\tx3401\\tx3968\\tx4535\\tx5102\\tx5669\\tx6236\\tx6803\\li720\\fi-720",
+        paste(
+          vapply(
+            seq_len(n_items),
+            FUN.VALUE = character(1),
+            USE.NAMES = FALSE,
+            FUN = function(n) {
+
+              paste0(
+                ifelse(n == 1, "\\ls1\\ilvl0\\cf0 \n", ""),
+                "{\\listtext\t}",
+                ifelse(type == "bullet", "\\u8226  ", ""),
+                process(xml2::xml_children(x)[n]),
+                "\\",
+                "\n"
+              )
+            }
+          ),
+          collapse = ""
+        ),
+        collapse = ""
+      )
+    )
+  },
+  item = function(x, process) {
+    # TODO: probably needs something like process_children()
+    rtf_escape(xml2::xml_text(x))
```

Comment:
What happens if you replace this with `process(xml2::xml_children(x))`?

Reply:
If you do make just that change, it just affects the markdown to RTF conversion of lists, and, we get unprocessed XML tags in the RTF.

With the following gt code

``` r
dplyr::tibble(markdown = "- item 1\n- item 2\n\n") %>%
  gt() %>%
  fmt_markdown(vars(markdown))
```

We get this RTF markup (just showing the one row):

```
\row

\trowd\trrh0

\pard\plain\uc0\ql\clvertalc\clpadb50\clpadfb3\clpadr50\clpadfr3\clpadl50\clpadfl3\clpadt50\clpadft3\clbrdrb\brdrs\brdrw10\brdrcf1\clbrdrl\brdrs\brdrw10\brdrcf1\clbrdrr\brdrs\brdrw10\brdrcf1\cellx9468
\intbl {\f0 {\f0\fs20 \pard\intbl\itap1\tx220\tx720\tx1133\tx1700\tx2267\tx2834\tx3401\tx3968\tx4535\tx5102\tx5669\tx6236\tx6803\li720\fi-720 \ls1\ilvl0\cf0 
{\listtext	}\u8226  <paragraph>
  <text xml:space="preserve">item 1</text>
</paragraph>\
{\listtext	}\u8226  <paragraph>
  <text xml:space="preserve">item 2</text>
</paragraph>\
}}\cell
\pard\plain

\row
```